tests/selenium/test_tooltip.py

```diff
@@ -0,0 +1,114 @@
+import dash
+import math
+import pytest
+from dash_table import DataTable
+
+columns = [dict(id=str(i), name="Column {}".format(i)) for i in range(1, 30)]
+
+data = []
+for i in range(1, 100):
+    datum = dict()
+    data.append(datum)
+    for j in columns:
+        datum[j["id"]] = "{}-{}".format(i, j["id"])
+
+tooltip_data_text = []
+for i in range(1, 100):
+    datum = dict()
+    tooltip_data_text.append(datum)
+    for j in columns:
+        datum[j["id"]] = dict(type="text", value=";; {}-{}".format(i, j["id"]))
+
+tooltip_data_markdown = []
+for i in range(1, 100):
+    datum = dict()
+    tooltip_data_markdown.append(datum)
+    for j in columns:
+        datum[j["id"]] = dict(type="markdown", value="### ;; {}-{}".format(i, j["id"]))
+
+base_props = dict(
+    id="table", columns=columns, data=data, tooltip_delay=None, tooltip_duration=None
+)
+
+
+def assert_aligned(cell, tooltip):
+    assert tooltip.location["x"] <= (cell.location["x"] + cell.size["width"])
+    assert (tooltip.location["x"] + tooltip.size["width"]) >= cell.location["x"]
```

Comment:
A tooltip should begin before or at the end of the cell and end after or at the beginning of the cell. More precise conditions are conditional on the tooltip size.